src/catalog/src/durable/persist.rs

```diff
@@ -1088,7 +1088,7 @@ impl UnopenedPersistCatalogState {
                 .into());
             }
             (
-                Mode::Writable,
+                Mode::Writable | Mode::Savepoint,
```

Comment:
Was it necessary to reintroduce this? Maybe this got pulled in as part of a rebase? Joe removed it very recently https://github.com/MaterializeInc/materialize/commit/76bdf33866ed305e7a1576c689f8cfb4f169a955

Reply:
Ah this was intentional—see https://github.com/MaterializeInc/materialize/pull/29792#pullrequestreview-2338601236!

Reply:
AFAICT, 76bdf33 was a hack to work around passing in the correct deploy generation, as we're setting up to do with this PR.

Reply:
Yes, I believe joe introduced this because it broke the upgrade checker... I fixed the upgrade checker :) 

https://github.com/MaterializeInc/materialize/pull/29792#pullrequestreview-2338601236

Reply:
Ah ha! Makes sense :)